src/openms/source/APPLICATIONS/TOPPBase.cpp

```diff
@@ -387,6 +408,130 @@ namespace OpenMS
       UniqueIdGenerator::setSeed(19991231235959);
     }
 
+    // enable / disable collection of usage statistics by build variable
+#ifdef ENABLE_USAGE_STATISTICS
+     bool collect_usage_statistics = true;
+#else
+     bool collect_usage_statistics = false;
+#endif
+
+    // disable collection of usage statistics if environment variable is present
+    char* env_usage = getenv("OPENMS_DISABLE_USAGE_STATISTICS");
+    if (env_usage != 0)
+    {
+      collect_usage_statistics = false;
+    }
+    /* debug code to print all environment variables
+    else
+    {
+      char *s = *environ;
+      int i = 1;
+      for (; s; i++) 
+      {
+        printf("%s\n", s);
+        s = *(environ+i);
+      }
+    }
+    */ 
+
+    if (!test_mode_ && collect_usage_statistics)
+    {
+      // if the revision info is meaningful, show it as well
+      String revision("UNKNOWN");
+      if (!VersionInfo::getRevision().empty() && VersionInfo::getRevision() != "exported")
+      {
+        revision = VersionInfo::getRevision();
+      }
+      String platform;
+
+      #ifdef OPENMS_WINDOWSPLATFORM
+        platform = "Win";
+      #elif __APPLE__
+        platform = "Mac";
+      #else
+        platform = "Linux";
+      #endif
+
+      String architecture = QSysInfo::WordSize == 32 ? "32" : "64";
+
+      // write to tmp + userid folder
+
+      // e.g.: OpenMS_Default_Win_64_FeatureFinderCentroided_2.0.0
+      String tool_version_string;
+      tool_version_string = String("OpenMS") + "_" + "Default_" + platform + "_" + architecture + "_" + tool_name_ + "_" + version_;
+
+      String version_file_name = File::getOpenMSHomePath() + "/.OpenMS/" + tool_name_ + ".ver";
```

Comment:
I really dislike this idea on a cluster, for example the home dir may not be mounted on each node. We already run into issues when trying to read the default config file for OpenMS. Is this covered by `File::readable`?